apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt

```diff
@@ -1,10 +1,117 @@
 package com.apollographql.apollo.api
 
-expect class ScalarTypeAdapters(customAdapters: Map<ScalarType, CustomTypeAdapter<*>>) {
+import com.apollographql.apollo.api.CustomTypeValue.*
+import com.apollographql.apollo.api.CustomTypeValue.Companion.fromRawValue
+import com.apollographql.apollo.api.internal.json.JsonWriter
+import com.apollographql.apollo.api.internal.json.Utils
+import com.apollographql.apollo.api.internal.json.use
+import okio.Buffer
+import kotlin.jvm.JvmField
 
-  fun <T : Any> adapterFor(scalarType: ScalarType): CustomTypeAdapter<T>
+class ScalarTypeAdapters(customAdapters: Map<ScalarType, CustomTypeAdapter<*>>) {
+  private val customAdapters = customAdapters.mapKeys { it.key.typeName() }
+
+  @Suppress("UNCHECKED_CAST")
+  fun <T : Any> adapterFor(scalarType: ScalarType): CustomTypeAdapter<T> {
+    var customTypeAdapter: CustomTypeAdapter<*>? = customAdapters[scalarType.typeName()]
+    if (customTypeAdapter == null) {
+      customTypeAdapter = DEFAULT_ADAPTERS[scalarType.className()]
+    }
+    return requireNotNull(customTypeAdapter) {
+      "Can't map GraphQL type: `${scalarType.typeName()}` to: `${scalarType.className()}`. Did you forget to add a custom type adapter?"
+    } as CustomTypeAdapter<T>
+  }
 
   companion object {
-    val DEFAULT: ScalarTypeAdapters
+    @JvmField
+    val DEFAULT = ScalarTypeAdapters(emptyMap())
+
+    private val DEFAULT_ADAPTERS = emptyMap<String, CustomTypeAdapter<*>>() +
+        createDefaultScalarTypeAdapter("java.lang.String", "kotlin.String") { value ->
```

Comment:
Is `java.lang.String` enough or should we also check other Strings? Like I wonder what would happen on iOS. Of course can be tackled later.

Reply:
on iOS it must be `kotlin.String` as generated code is platform agnostic (common).

Reply:
👍 sounds good.